### ISSUE TITLE
Wrong code generation with multipart files

### ISSUE THREAD

**Issue by iorran** [anymaniax/orval#480]
## What are the steps to reproduce this issue?

1. Using React query generator option
2. When desire multiple documents to be sent once

## What happens?

Orval is generating a method that wraps all the files in JSON.stringfy

## What were you expecting to happen?

Instead of 

```js
export const uploadMultipleDocuments = (
  uploadMultipleDocumentsBody: UploadMultipleDocumentsBody,
  options?: SecondParameter<typeof customInstance>
) => {
  const formData = new FormData()
  formData.append('files', JSON.stringify(uploadMultipleDocumentsBody.files))
  formData.append('submissionName', uploadMultipleDocumentsBody.submissionName)

  return customInstance<void>(
    {
      url: `/documents/multiple`,
      method: 'post',
      headers: { 'Content-Type': 'multipart/form-data' },
      data: formData,
    },
    options
  )
}
```

### We might have

```js
export const uploadMultipleDocuments = (
  uploadMultipleDocumentsBody: UploadMultipleDocumentsBody,
  options?: SecondParameter<typeof customInstance>
) => {
  const formData = new FormData()
  uploadMultipleDocumentsBody.files.forEach(file -> formData.append('files', file))
  formData.append('submissionName', uploadMultipleDocumentsBody.submissionName)

  return customInstance<void>(
    {
      url: `/documents/multiple`,
      method: 'post',
      headers: { 'Content-Type': 'multipart/form-data' },
      data: formData,
    },
    options
  )
}
```

## Any other comments?

If I'm doing something wrong with my specification, please, help me out.

```yml
/documents/multiple:
    post:
      tags:
        - document-controller
      operationId: uploadMultipleDocuments
      description: Uploads multiple documents to the server
      parameters:
        - in: header
          name: X-Request-ID
          required: true
          schema:
            $ref: '#/components/schemas/XRequestIDHeader'
        - in: header
          name: X-Platform
          required: true
          schema:
            $ref: '#/components/schemas/XPlatformHeader'
        - in: header
          name: X-Version
          required: true
          schema:
            $ref: '#/components/schemas/XVersionHeader'
        - in: header
          name: Authorization
          required: true
          schema:
            type: string
      security:
        - jwt: ['secret']
      requestBody:
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                files:
                  type: array
                  items:
                    type: string
                    format: binary
                submissionName:
                  type: string
                  minLength: 1
              required:
                - files
                - submissionName
      responses:
        '200':
          description: OK
        '400':
          description: Invalid user request
        '401':
          description: Unauthorized
        '403':
          description: User not allowed to perform the requested action
        '500':
          description: Internal server error
```

## What versions are you using?

**Operating System:** MacOS
**Package Version:** "orval": "^6.8.1"

**Comment by iorran** [comment]
I found a way to override it, but, I still think this should be the default behavior.

Orval config: 
```js
module.exports = {
  'v2/mw-api': {
    output: {
      mode: 'split',
      target: './src/services/v2/mw-api.ts',
      schemas: './src/services/v2/model',
      client: 'react-query',
      mock: true,
      prettier: true,

      override: {
        mutator: {
          path: './src/services/custom-instance.ts',
          name: 'customInstance',
        }, 
        operations: {
          uploadMultipleDocuments: { 
            formData: {
              path:  './src/utils/custom-upload-multiple-documents.ts',
              name: 'customUploadMultipleDocuments',
            },
          }
        }
      },
    },
    input: {
      target: './src/services/open-api/mobile_api_v2.yaml',
    },
    hooks: {
      afterAllFilesWrite: 'prettier --write',
    },
  },
}
```
Function to override:
```js
export const customUploadMultipleDocuments = (param: {files: Blob[], submissionName: string}): FormData => {
  const formData = new FormData()

  param.files.forEach(file => formData.append('files', file))
  formData.append('submissionName', param.submissionName)

  return formData;
};
```
Result:

```js
/**
 * Uploads multiple documents to the server
 */
export const uploadMultipleDocuments = (
  uploadMultipleDocumentsBody: UploadMultipleDocumentsBody,
  options?: SecondParameter<typeof customInstance>
) => {
  const formData = customUploadMultipleDocuments(uploadMultipleDocumentsBody)
  return customInstance<void>(
    {
      url: `/documents/multiple`,
      method: 'post',
      headers: { 'Content-Type': 'multipart/form-data' },
      data: formData,
    },
    options
  )
}
```

**Comment by anymaniax** [comment]
will be added in the next release

**Comment by anymaniax** [comment]
Released with 6.9.0
